### PR TITLE
feat!:try/with, string operations, primes

### PR DESCRIPTION
**Pull request by SpiralOSS**

### CODE REVIEW

File: docs/fsharp-cheatsheet.md
Comment (bartelink) [inline docs/fsharp-cheatsheet.md:74]:
```suggestion
See [Strings (MS Learn)](https://learn.microsoft.com/en-us/dotnet/fsharp/language-reference/strings) for more on escape characters, byte arrays, and format specifiers.
```

File: docs/fsharp-cheatsheet.md
Comment (bartelink) [inline docs/fsharp-cheatsheet.md:384]: any reason for extra indent?

File: docs/fsharp-cheatsheet.md
Comment (bartelink) [inline docs/fsharp-cheatsheet.md:393]: should there be a space before `(str)` ?  (For a DU I would add one)
(but I'd probably remove the parens too)
Agree with the line break

File: docs/fsharp-cheatsheet.md
Comment (bartelink) [inline docs/fsharp-cheatsheet.md:465]: I dont like the then on a new line (I'd join the if and then lines)
but it probably is good to show it this way pedagogically

File: docs/fsharp-cheatsheet.md
Comment (bartelink) [inline docs/fsharp-cheatsheet.md:382]:
```suggestion
The `try/finally` expression enables you to execute clean-up code even if a block of code throws an exception. Here's an example that also defines custom exceptions.
```

**Review by bartelink — APPROVED** [review]
All looks good - will merge when resolved

File: docs/fsharp-cheatsheet.md
Comment (SpiralOSS) [reply docs/fsharp-cheatsheet.md:95]: I can be convinced otherwise, but I didn't think it was a good fit for this. It all seems a little too obvious.

File: docs/fsharp-cheatsheet.md
Comment (SpiralOSS) [reply docs/fsharp-cheatsheet.md]: fixed

File: docs/fsharp-cheatsheet.md
Comment (SpiralOSS) [reply docs/fsharp-cheatsheet.md]: fixed

**Comment by SpiralOSS** [comment]
@bartelink I named this PR "Updates"; and I noticed on our last PR, you changed the name and prefixed with "feat!" and named it slightly different. Is there some doco on naming practices?

**Comment by bartelink** [comment]
> Is there some doco on naming practices

https://www.conventionalcommits.org/en/v1.0.0/

Very annoying/wierd for about 2 days, then you can't stop yourself

TL;DR, composed of
task code (e.g.: feat/chore/fix)
(area) (optional)
! (if breaking)
: 
description that does not repeat any of the previous

- feat(Quotes): Explain triple quotes
- fix(Patterns): Handle malformed inputs
- feat(Patterns)!: Remove usage of fst/snd
- chore: update runner versions

In the context of code, the `!` (breaking) indicator would mean a semver interface change or a dependency on an new mandatory env var

In the context of this repo, `!` would mean a subjective rewrite of something 

In prod code:
chore(deps): Update json libs
fix(Profile): Render name correctly
fix(Host)!: Require DB server name via DB_SERVER_NAME

File: docs/fsharp-cheatsheet.md
Comment (bartelink) [reply docs/fsharp-cheatsheet.md:95]: I think it's ok to have in an appendix - it definitely is something JS or CS people won't have seen so saying "this thing is not special, it's just a diff char that F# allows in the name" is of value IMO

and it can live alongside triple backticks - people never shut up about it in intros but its only used in some places. saying "used for test names, probably should not be used willy nilly" is helpfuy;

BUT, in here, it's distracting to the point being made, so a footnote ref would be the ideal representation

**Comment by bartelink** [comment]
> I named this PR "Updates";

conventional commits also nudges toward SRP for PRs - i.e. there's no good name for this grab bag of stuff (not saying it needs to be split as long as its not in flight for long and you are the maintainer after all!)

File: docs/fsharp-cheatsheet.md
Comment (SpiralOSS) [reply docs/fsharp-cheatsheet.md:473]: Interesting. Something to consider for my own practices. I'll leave the underscore as one of those pedagogical examples.

File: docs/fsharp-cheatsheet.md
Comment (bartelink) [inline docs/fsharp-cheatsheet.md:462]: not a great example as you'd really want this to either be a partial or have an explicit non matcing one
also I think the Integer and Regex examples in https://learn.microsoft.com/en-us/dotnet/fsharp/language-reference/active-patterns#parameterized-active-patterns might be useful if we were to give some more space to patterns

**Pull request merged**
